templates/simp_generate_types_incron_rules.epp

```diff
@@ -0,0 +1,17 @@
+<%- |
+  Boolean                     $trigger_on_puppetserver_update,
+  Boolean                     $trigger_on_puppet_update,
+  Array[Stdlib::AbsolutePath] $trigger_paths,
+  Stdlib::AbsolutePath        $simp_generate_types
+| -%>
+<% if $trigger_on_puppetserver_update { -%>
+/opt/puppetlabs/server/apps/puppetserver/bin IN_MODIFY,IN_NO_LOOP <%= $simp_generate_types %> -p all -s
+<% } -%>
+<% if $trigger_on_puppet_update { -%>
+/opt/puppetlabs/puppet/bin/puppet IN_MODIFY,IN_NO_LOOP <%= $simp_generate_types %> -p all -s
+<% } -%>
+<%
+  $trigger_paths.each |Stdlib::AbsolutePath $path| {
+-%>
+<%= $path %> IN_MODIFY,IN_CREATE,IN_NO_LOOP <%= $simp_generate_types %> -p $@/$# -s
```

Comment:
Does IN_NO_LOOP disable monitoring of ALL events, only events specified in a user's incrontab, or only events related to the file specified?  If the last option, it doesn't seem like it will be real useful.

I do not see anything that guarantees all the files for a type are laid down before the puppet types generation is called.

Reply:
`IN_NO_LOOP` disables monitoring of all events in that incrontab.

Being unable to see into the future, I'm adding a `delay` option that will be set to `30 seconds` by default. I've run `puppet generate types` in four sessions simultaneously and have not seen any adverse reactions so other scripts calling the procedure should not cause any issues.